pkg/sql/opt_catalog.go

```diff
@@ -1821,6 +1824,7 @@ func newOptVirtualTable(
 			cat.MaybeHidden(d.IsHidden()),
 			d.ColumnDesc().DefaultExpr,
 			d.ColumnDesc().ComputeExpr,
+			d.ColumnDesc().OnUpdateExpr,
```

Comment:
I'm aware this is continuing an existing pattern but we _really_ should avoid looking stuff up in `ColumnDesc()` or `IndexDesc()` when possible and instead delegate to the `catalog.Column` interface. 

Perhaps this is out of scope for this change. Also I'm aware that the opt-catalog predates the schema-catalog but at some point we'll have to do this change so might as well start now and make it happen gradually?